homeassistant/components/fritz/button.py

```diff
@@ -90,7 +90,8 @@ def __init__(
         self.router = router
 
         self._attr_name = f"{device_friendly_name} {description.name}"
-        self._attr_unique_id = slugify(self._attr_name)
+        self._attr_unique_id = f"{self.router.unique_id}-{slugify(description.name)}"
```

Comment:
Why does the router have a mac property and a unique_id property that both return the same thing?

Reply:
Use the key instead of the description name. Then we don't need to slugify.

Reply:
Done